NuGet/NuGet.Client
# Change Apex tests from Xunit to MSTest

## Bug



Fixes: https://github.com/nuget/client.engineering/issues/852

Regression? Last working version:

## Description
**Background**
Since Xunit is not able to timeout individual test ([comment](https://github.com/nuget/client.engineering/issues/852#issuecomment-875681103)), in order to time out individual Apex tests, change Apex tests from Xunit to MSTest.

1.Change Apex test from Xunit to MSTest:
   * Change Xunit attributes to MSTest attributes: `Theory `-> `TestMethod`, `InlineData `-> `DataRow`, `Skip `-> `Ignore`, etc.
   * Add `Timeout `attribute for each Apex tests
   * Change Xunit.Assert to MSTest assertion 
   * Adapt setup of Xunit (fixture) to MSTest setup (ClassInitialize). (MSTest has [limitation ](https://github.com/microsoft/testfx/issues/143) as ClassInitialize is not inheritable. So change the previous fixture for base test class to a static property.)
   * Change Xunit log into Trace.WriteLine. (a test [build ](https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=5326490&view=ms.vss-test-web.build-test-results-tab&runId=27181036&resultId=100007&paneView=attachments) shows that when the test failed, the log will be shown)
 2. Engineering change 
 * Add package for MSTest: MSTest.TestAdapter and MSTest.TestFramework
 * Change build/template.runsettingsproj to add console logger, so that trace will be shown in the log.
 * Clean up test/NuGet.Tests.Apex/NuGet.Tests.Apex/xunit.runner.json
 * Decide not to clean up xunit packages from Apex test projects, as Apex test projects reference Test.Utility project, which has xunit package reference. We will have to get rid of all references to xunit from Test.Utility project first. I don't think it's worth doing compared to the effort (The only impact is we still see several xunit message in test logs).

## PR Checklist


- **Tests**
  - **OR**
  
  - **OR**

- **Documentation**
  
  - **OR**
